File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
   
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int i=0;
        int start=true;
        int currentRun=0;
        while(i<flowerbed.size()){
            if(flowerbed[i]==0){
                currentRun++;
            }
            else{
                if(currentRun>=2 and start==true){
                    n-=currentRun/2;
                }
                else if(currentRun>=3){
                    currentRun%2==0?n-=currentRun/2-1:n-=currentRun/2;
                }
                start=false;
                currentRun=0;
            }
            i++;
        }
        if(start==true){
            currentRun%2==1?currentRun+=1:currentRun=currentRun;
            cout<<currentRun;
            n-=currentRun/2;
        }
        else if(currentRun>=2){
            n-=currentRun/2;
        }
        if(n<=0){
            return true;
        }
        return false;
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        ##Start from the beginning
        av= 0 
        if len(flowerbed)==1 and flowerbed[0]==0 and n==1:
            return True
        for i in range(len(flowerbed)):
            if i==0 and len(flowerbed)>1 and flowerbed[i+1]==0 and flowerbed[i]==0:
                flowerbed[0]= 1
                av+=1
            elif i==len(flowerbed)-1 and len(flowerbed)>1 and flowerbed[-2]==0 and flowerbed[i]==0:
                flowerbed[-1]=1
                av+=1
            elif i-1>0 and i+1<len(flowerbed) and flowerbed[i-1]==0 and flowerbed[i]==0 and flowerbed[i+1]==0:
                flowerbed[i]=1
                av+=1        
        if av>=n:
            return True
        return False